onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1537,7 +1538,7 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<Node*>& fuse
 
         // Build engine
         {
-          auto lock = GetEngineBuildLock();
+          auto lock = GetApiLock();
```

Comment:
I was wondering do we need a lock here? since the whole inference has been protected by a lock already.
https://github.com/microsoft/onnxruntime/blob/a2c14ca8b7010dc414b2033610372f1483c20836/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc#L1290